### PR TITLE
Add a restore command

### PR DESCRIPTION
**Pull request by algrs**
to restore deleted files from versioned buckets.

### CODE REVIEW

File: baiji/cli.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ class ListCommand(BaijiCommand):
     DESCRIPTION = "list files on s3"
     uri = cli.Flag(["-B", "--uri"], help='This option does nothing. It used to return URIs instead of paths, but this is now the default.')
     detail = cli.Flag(['-l', '--detail'], help='print details, like `ls -l`')
-    shallow = cli.Flag("--shallow", help='process key names hierarchically and return only immediate "children" (like ls, instead of like find)')
+    shallow = cli.Flag(['-s', "--shallow"], help='process key names hierarchically and return only immediate "children" (like ls, instead of like find)')
     list_versions = cli.Flag(['--list-versions'], help='print all versions')
```
Comment (lillekemiker) [inline baiji/cli.py]: hahaha, after all these years, you finally got sick of typing out '--shallow'?? :)

File: baiji/util/console.py [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+            style = Default
+        self.spinner = Spinner(style)
+        self.last_len = 0
+    def spin(self, text):
+        self.display(u"{} {}".format(self.spinner.next(), text))
+    def drop(self, text):
+        self.display(u"{}\n".format(text))
+    def display(self, text):
+        print(u'\r{}{}'.format(text, " " * (self.last_len - len(text))), end='')
+        self.last_len = len(text)
```
Comment (lillekemiker) [inline baiji/util/console.py]: `sys.stdout.flush()`?

File: baiji/cli.py [hunk 3]
```diff
@@ -39,6 +39,20 @@ def main(self, key):
...
+        from baiji.util.console import LabeledSpinner
+        if s3.path.islocal(prefix):
+            raise ValueError("restore command only works on s3")
+        spin = LabeledSpinner()
+        for key in s3.ls(prefix, return_full_urls=True, require_s3_scheme=True, list_versions=True):
+            if not s3.exists(key):
+                spin.drop("Restoring deleted file {}".format(key))
+                s3.restore(key)
+            else:
+                spin.spin(key)
```
Comment (lillekemiker) [inline baiji/cli.py]: Makes me want to try recovering files :)

[Edit: I just realize that I really only get the spinner if the files don't need recovering... so I can totally try this without actually recovering deleted files :D ]

File: baiji/util/console.py (hunk 2 above)
Comment (algrs) [inline baiji/util/console.py]: Sure

**Pull request merged**